lib/membrane_audio_mixer.ex

```diff
@@ -292,4 +331,18 @@ defmodule Membrane.AudioMixer do
     end)
     |> Map.new()
   end
+
+  defp mix_payloads(payloads, %{prevent_clipping: %ClipPreventer.State{} = preventer} = state) do
+    all_ended = all_streams_ended?(state)
+
+    {payload, preventer} = ClipPreventer.mix(payloads, all_ended, state.caps, preventer)
+    state = %{state | prevent_clipping: preventer}
+
+    {payload, state}
+  end
+
+  defp mix_payloads(payloads, state) do
+    payload = DoMix.mix(payloads, state.caps)
+    {payload, state}
+  end
```

Comment:
This is exactly what we wanted to avoid by splitting mixers to separate modules.

1. Define behaviour for mixers with `init`, `mix` and `flush`
2. Make sure both modules implement that
3. Store the mixer module name & state from init in the element state - you can use the structs for that, ensure init returns a struct even if it's empty
4. Call the proper functions dynamically:
```elixir
%mixer_module{} = state.mixer_state
mixer_module.mix([], state)
```